src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/Tasks/CompilerState.cs

```diff
@@ -251,6 +251,31 @@ private static string GenerateStringFromFileNames(ITaskItem[] fileItemList)
             return fileNames;
         }
 
+        //
+        // Generates a stable hash code value for strings.
+        // In .NET Core the hash values for the same string can be different between
+        // subsequent program runs and cannot be used for caching here.
+        // source: https://andrewlock.net/why-is-string-gethashcode-different-each-time-i-run-my-program-in-net-core/#a-deterministic-gethashcode-implementation
+        //
+        private static int GetDeterministicHashCode(string str)
```

Comment:
@MichaeIDietrich - your fix is the correct fix. Can I ask that we instead use the internal String.Comparison method from .NET. Feel free to copy paste: https://source.dot.net/#System.Private.CoreLib/String.Comparison.cs,807

Thank you!

        internal unsafe int GetNonRandomizedHashCode()
        {
            fixed (char* src = &_firstChar)
            {
                Debug.Assert(src[this.Length] == '\0', "src[this.Length] == '\\0'");
                Debug.Assert(((int)src) % 4 == 0, "Managed string should start at 4 bytes boundary");
 
                uint hash1 = (5381 << 16) + 5381;
                uint hash2 = hash1;
 
                uint* ptr = (uint*)src;
                int length = this.Length;
 
                while (length > 2)
                {
                    length -= 4;
                    // Where length is 4n-1 (e.g. 3,7,11,15,19) this additionally consumes the null terminator
                    hash1 = (BitOperations.RotateLeft(hash1, 5) + hash1) ^ ptr[0];
                    hash2 = (BitOperations.RotateLeft(hash2, 5) + hash2) ^ ptr[1];
                    ptr += 2;
                }
 
                if (length > 0)
                {
                    // Where length is 4n-3 (e.g. 1,5,9,13,17) this additionally consumes the null terminator
                    hash2 = (BitOperations.RotateLeft(hash2, 5) + hash2) ^ ptr[0];
                }
 
                return (int)(hash1 + (hash2 * 1566083941));
            }
        }

Reply:
The `BitOperations` class is not available for the used TFMs. Do you have any preferences for inlining the left rotation code here?

Reply:
No preference.  Anything is better than what we had before.